src/vma/dev/ring.h

```diff
@@ -80,27 +80,56 @@ typedef enum {
 
 /* udp uc key, only by destination port as we already know the rest */
 typedef struct __attribute__((packed)) flow_spec_udp_uc_key_t {
-	in_port_t 	dst_port;
-	flow_spec_udp_uc_key_t(): dst_port(INPORT_ANY){}//Default constructor
-	flow_spec_udp_uc_key_t(in_port_t d_port): dst_port(d_port){}//Constructor
+  in_port_t 	dst_port;
+
+  /* Just for silencing coverity */
+  flow_spec_udp_uc_key_t () {
+    memset(this, 0, sizeof(*this));
+    dst_port = INPORT_ANY;
+  } //Default constructor
+  flow_spec_udp_uc_key_t (in_port_t d_port) {
+    memset(this, 0, sizeof(*this));
+    dst_port = d_port;
+  }//Constructor
 } flow_spec_udp_uc_key_t;
 
 typedef struct __attribute__((packed)) flow_spec_udp_mc_key_t {
-	in_addr_t	dst_ip;
-	in_port_t	dst_port;
-	flow_spec_udp_mc_key_t(): dst_ip(INADDR_ANY), dst_port(INPORT_ANY){}//Default constructor
-	flow_spec_udp_mc_key_t(in_addr_t d_ip, in_addr_t d_port): dst_ip(d_ip), dst_port(d_port){}//Constructor
+  in_addr_t	dst_ip;
+  in_port_t	dst_port;
+
+  /* Just for silencing coverity */
+  flow_spec_udp_mc_key_t () {
+    memset(this, 0, sizeof(*this));
```

Comment:
will calling `flow_spec_udp_mc_key_t (INADDR_ANY, INPORT_ANY)` work as well?
this can reduce amount of duplicated code.